lib/order.rb

```diff
@@ -0,0 +1,99 @@
+require 'csv'
+require_relative 'customer'
+
+class Order
+  attr_reader :id
+  attr_accessor :products, :customer, :fulfillment_status
+
+  @@all_orders = []
+
+  # Constructor method to initialize all Order.new instances
+  def initialize(id, products, customer, fulfillment_status = :pending)
+    @id = id.to_i
+    @products = products
+    @customer = customer
+    @fulfillment_status = fulfillment_status
+
+    # Raise ArgumentError if fulfillment_status is not one of five options
+    @fulfillment_options = [:pending, :paid, :processing, :shipped, :complete]
+    unless @fulfillment_options.include? @fulfillment_status
+      raise ArgumentError, 'Fulfillment status must be :pending, :paid, :processing, :shipped, or :complete. '
+    end
+  end
+
+  # Calculate total of Order instance products
+  def total
+    # if @products.empty, product sum is 0
+    if @products == {} || @products == 0
+      @product_sum = 0
+    else
```

Comment:
I think you could get away without this conditional. If there are no products, then the inside of your `@products.each_value` loop will execute exactly 0 times, resulting in a total of 0.